packages/components/src/DataList/components/DataListLayoutInternal/DataListItemInternal.tsx

```diff
@@ -0,0 +1,39 @@
+import classNames from "classnames";
+import React from "react";
+import { Checkbox } from "../../../Checkbox";
+import { useDataListContext } from "../../context/DataListContext";
+import styles from "../../DataList.css";
+import { DataListObject } from "../../DataList.types";
+
+export function ListItemInternal<T extends DataListObject>({
+  children,
+  item,
+}: {
+  children: JSX.Element;
+  item: T;
+}) {
+  const { selected, onSelect } = useDataListContext();
+
+  if (typeof selected !== "undefined" && onSelect) {
```

Comment:
that an interesting approach. Out of curiosity, why does checking `selected !== undefined` not working?

Reply:
Honestly, I think it was a small remnant of when you compare types for narrowing you do this but undefined works. There is some stuff related to undeclared variables with it but that isn't a concern here
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined#typeof_operator_and_undefined